src/helpers/momentHelper.js

```diff
@@ -74,3 +74,56 @@ export const isActive = (item) => {
 
   return true;
 };
+
+/**
+ *
+ * Find the item with the closest date range to today.
+ *
+ * @param {DataItem[]} data an array of objects, each with a `dates` array
+ *
+ * @return {DataItem | null} the item with the closest date range to today, or null if no such item exists
+ */
+export const findClosestItem = (data) => {
+  // Check if data is null or undefined or not an array
+  if (!Array.isArray(data)) {
+    return null;
+  }
+
+  const today = new Date();
+
+  return data.reduce(
+    (closest, item) => {
+      // Check if item or item.dates is null or undefined or not an array
+      if (!item || !Array.isArray(item.dates)) {
+        return closest;
+      }
+
+      // Calculate the minimum time difference between today and the date ranges in the current item.
+      const minDifference = Math.min(
+        ...item.dates.map((dateRange) => {
+          // Check if dateRange, dateStart, or dateEnd is null or undefined
+          if (!dateRange || !dateRange.dateStart || !dateRange.dateEnd) {
+            return Infinity;
+          }
+
+          // Convert the start and end dates to time in milliseconds.
+          const dateStart = moment(dateRange.dateStart).valueOf();
+          const dateEnd = moment(dateRange.dateEnd).valueOf();
+
+          // Calculate the absolute difference between today and both the start and end dates,
+          // and return the smaller of the two.
+          return Math.min(
+            Math.abs(today.getTime() - dateStart),
```

Comment:
`today` is only needed here, right? so you could move `const today = new Date();` right before this return, to avoid having the variable created in all the other cases, where it is not needed.